cocotb/regression.py

```diff
@@ -617,21 +618,21 @@ def _create_test(function, name, documentation, mod, *args, **kwargs):
     Returns:
         Decorated test function
     """
+    # causes early type check, during call to generate_tests()
+    test = cocotb.scheduler.create_task(function(cocotb.top, *args, **kwargs))
```

Comment:
I think it's better to leave the task creation as part of the Regression Manager, so we have one place that the root handle is passed in. If you want the check at this point, you could create the coroutine object here but not keep it.

Reply:
The issue if that the callable may have side effects. Calling it more than once may not be permissible. I'm not against simply removing the check. `cocotb.test()` does not check the type is correct until the test is collected, I don't see why `TestFactory` should have ever been different (I think I was the one who added the checks so feel free to call me inconsistent).